### ISSUE TITLE
Problems building aspell with Forte compiler

### ISSUE THREAD

**Issue by aspell-helper** [GNUAspell/aspell#353]
**Brian Cameron *<yippi\@sf>* created a bug report on 2006-01-25 23:34:08 UTC**
*(Orig. from https://sourceforge.net/p/aspell/bugs/178)*

The attached patch resolves a number of issues building
aspell with the Forte compiler.  

Issues include:

1\) Forte doesn't like it when there are structures
where the variable has the same name as the type. 
So lines like this cause problems:

typedef Vector<u32int>      Vector;

This generates "Redefining x after use" errors.

There were two such problems in the code, another 
with "hash".  I renamed these to TheVector and 
thehash and this corrected the problem.  The 
maintainer might have better names for the variable
but the patch shows how to fix the problem.

2\) In modules/speller/default/affix.hpp there are two
variables in the argument list with "new\_aff" as the
name.  Forte doesn't like this.  Named the first one
as "aff" since that is what is in the prototype 
definition for the function.

3\) Forte doesn't like "static inline" functions defined
in header files, so common/string.hpp has been 
fixed to remove the "static" keyword.

4\) The "static" used by free\_norm\_table in    
common/convert.cpp was making Forte complain that it
couldn't use the function for the places where it
is used.  Removing the "static" fixed this.

5\) Forte doesn't like doing things like 
address = address + sizeof\(variable\)/sizeof\(struct\)
when the variable is empty.  This was causing 
problems in the code generated by 
mk-static-filter.pl because url\_options is empty.
In other words, it looks like this:

static KeyInfo url\_options\[\] = \{

\};

const KeyInfo \* url\_options\_begin = url\_options;

const KeyInfo \* url\_options\_end = url\_options +
sizeof\(url\_options\)/sizeof\(KeyInfo\);

This caused problems because the Forte does 
not like doing sizeof\(url\_options\) when it is 
empty.

In this case, sizeof\(url\_options\)/sizeof\(KeyInfo\)
is 0, so I modified the script so that if the
size of the structure is zero, it just puts this
in the file:

const KeyInfo \* url\_options\_end = url\_options;

Which makes Forte happier and is the same
value anyway.
